src/core/CMakeLists.txt

```
@@ -921,6 +921,22 @@ if (ENABLE_WEB_SERVICE)
     target_link_libraries(core PRIVATE web_service)
 endif()
 
+if (ARCHITECTURE_arm64)
```

Linux-only for the time being
```suggestion
if (ARCHITECTURE_arm64 AND UNIX AND NOT APPLE)
```
(might need to be (UNIX OR ANDROID), not sure)

---

Maybe AND NOT WIN32 AND NOT APPLE?

---

how about this: in the root CMakeLists, add
```
if (ARCHITECTURE_arm64 AND (ANDROID OR ${CMAKE_SYSTEM_NAME} STREQUAL "Linux"))
    set(HAS_NCE ON)
    add_compile_definitions(HAS_NCE)
endif()
```
and then just use that for the Linux-specific cases where we are currently checking for ARCHITECTURE_arm64. I will have to do a mac build at some point to make sure.